File: src/sac_lib_errortest.rb
$dependencies << 'character_lib.o'
$dependencies << 'sac_lib.o'
$compiler = ENV.fetch('MY_FORTRAN_DEBUG', 'gfortran -ffree-line-length-none -fmax-identifier-length=63 -pipe -cpp -C -Wall -fbounds-check -O0 -fbacktrace -ggdb -pg -DDEBUG')

setup <<-EOS
# include "../fortran_lib.h"
program runner
  USE_FORTRAN_LIB_H
  use, non_intrinsic:: sac_lib

  implicit none

  type(Sac):: wHead
  Character:: cTrash
  Integer:: iTrash
EOS

errortest 'set_kstnm with too long argument', "call set_kstnm(wHead, '0123456789abcdefg')"
errortest 'set_kevnm with too long argument', "call set_kevnm(wHead, '123456789')"
errortest 'set_iftype with invalid argument', "call set_iftype(wHead, 'anything other than enumerated values')"

errortest 'get_iftype for undefined value', "cTrash = get_iftype(wHead)"
errortest 'get_imagsrc for undefined value', "cTrash = get_imagsrc(wHead)"

teardown <<-EOS
  stop
end program runner
EOS

File: sac_lib_errortest.f90
# include "fortran_lib.h"
program main
   USE_FORTRAN_LIB_H
   use, intrinsic:: iso_fortran_env, only: input_unit, output_unit
   use, non_intrinsic:: sac_lib

   implicit none


   Integer:: i


   read(input_unit, *) i


   select case(i)
   case(1)
      ! kevnm with too long argument
      block
         type(Sac):: wHead
         call set_kevnm(wHead, '123456789abcdefgh')
      end block
   case(2)
      ! kstnm with too long argument
      block
         type(Sac):: wHead
         call set_kstnm(wHead, '0123456789abcdefg')
      end block
   case(3)
      ! iftype with invalid argument
      block
         type(Sac):: wHead
         call set_iftype(wHead, 'anything other than enumerated values')
      end block
   case(4)
      ! imagsrc for undefined value
      block
         type(Sac):: wHead
         Character:: cTrash
         cTrash = get_imagsrc(wHead)
      end block
   case(5)
      ! iftype for undefined value
      block
         type(Sac):: wHead
         Character:: cTrash
         cTrash = get_iftype(wHead)
      end block
   case default
      write(output_unit, *) 4
   end select


   stop
end program main
